masterfabric-mobile/swift-camp
# CL - LongPressGesture Viper Implementation

## 📋 PR Description


This page provides an interactive guide to the versatile use of longPressGesture in SwiftUI, showcasing various scenarios. From simple long-press actions to advanced combinations like object scaling, rotation, and dragging, it covers a wide range of examples enriched with animations and user-friendly guidance messages. Designed to inspire both beginners and advanced developers, this page makes creating gesture-driven user interfaces in SwiftUI easier and more engaging!

Additionally, this page now includes a map interaction feature, allowing users to select a location on the map. Initially, the map displays pins for San Francisco and the Golden Gate Bridge. Users can long-press on the map to select a custom location and explore the map interactively. This enhancement simplifies the user experience for map interactions and location selection, making it more intuitive and user-friendly.

---

## ✅ Checklist


---

## 🛠 Steps to Test



1. Open the project in Xcode and run the app on a simulator or a physical device.
2. Navigate to the “Long Press Gesture Examples” page.
3. Test Example 1 by long-pressing the button for 1 second and ensuring the text toggles between “Long Press Me” and “Pressed!”
4. Test Example 2 by holding down for 2 seconds and verifying the feedback messages (“Keep holding…” and “Hold complete!”) display correctly.
5. Test Example 3 by either long-pressing for 1 second or double-tapping the text and confirming the appropriate message (“Long Press Detected!” or “Double Tap Detected!”) appears.
6. Test Example 4 by long-pressing the circle for 1.5 seconds and ensuring it scales up and down smoothly.
7. Test Example 5 by long-pressing the rectangle for 1 second and verifying it rotates 45 degrees on each press.
8. Test Example 6 by dragging the blue rectangle to a new position and confirming that long-pressing it resets its position to the original location.
9. Ensure all animations and gestures work seamlessly without any errors or lag.
10. Launch the application and navigate to the map view.
11. Verify that the map displays the initial pins for San Francisco and the Golden Gate Bridge.
12. Long-press anywhere on the map to select a new location.
13. Ensure that the coordinates of the selected location are displayed correctly on the screen.
14. Test the interaction multiple times to ensure the functionality works consistently each time.

---

## 🔗 Related Links

- Documentation: 
- LongPressGesture : https://developer.apple.com/documentation/swiftui/longpressgesture
- Map : https://developer.apple.com/documentation/mapkit/map

---

### 📷 Screenshots (Optional)


<img width="304" alt="Ekran Resmi 2024-12-29 23 31 37" src="https://github.com/user-attachments/assets/dab67c32-5cbf-4693-a0fe-7a98e2bebad9" />
<img width="304" alt="Ekran Resmi 2024-12-29 23 31 46" src="https://github.com/user-attachments/assets/246688a8-de49-4f6b-9aec-9a456c79756f" />
<img width="304" alt="Ekran Resmi 2024-12-29 23 32 07" src="https://github.com/user-attachments/assets/df9ac5e0-9c28-47d7-add2-8a23e1c3ae33" />
<img width="304" alt="Ekran Resmi 2024-12-29 23 32 17" src="https://github.com/user-attachments/assets/234f0bfe-e15e-4de5-95c0-60feae21823c" />
<img width="304" alt="Ekran Resmi 2024-12-29 23 33 03" src="https://github.com/user-attachments/assets/1c105314-3bea-4c01-b4bd-8a920fe44fbf" />
